std/datetime/systime.d

```diff
@@ -8414,18 +8414,19 @@ public:
     {
         import std.algorithm.searching : countUntil, find;
         import std.conv : to;
-        import std.string : strip;
+        import std.string : strip, indexOf;
+        import std.utf : byCodeUnit;
 
-        auto dstr = to!dstring(strip(isoExtString));
+        auto str = strip(isoExtString);
 
-        auto tIndex = dstr.countUntil('T');
+        auto tIndex = str.indexOf('T');
         enforce(tIndex != -1, new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
 
-        auto found = dstr[tIndex + 1 .. $].find('.', 'Z', '+', '-');
-        auto dateTimeStr = dstr[0 .. $ - found[0].length];
+        auto found = str[tIndex + 1 .. $].find('.', 'Z', '+', '-');
```

Comment:
Isn't auto-decoding happening here?

Reply:
For reference:

https://github.com/dlang/phobos/blob/12a2f9b874db8b8ade49adcd6ee4550055c48d36/std/algorithm/searching.d#L1559-L1574